app/models/sources/joyent.rb

```diff
@@ -26,6 +26,24 @@ def validate_config!
       return false
     end
 
+    # Connnect to Joyent and find all of the machines with that combination of
+    # user and datacenter that is provided with each source object source.
+    # Each machine found will be delete.
+    def purge_servers(captcha=nil)
+      if captcha.nil?
+        puts("What is 1 + 2 + 3 + 4? Put answer as a parameter.")
+      elsif captcha == 10
+        connection.servers.select do |s|
```

Comment:
`.each` is a better choice since I'm not providing a condition with `.select`. Will make change.

Moving this code to be in Fog also makes sense since I am only using Fog functions at the moment.

Deleting both proxies and non-proxies was what was decided in the original discussion to have this function do. I can set it up so that it can delete only proxies, only non-proxies, and both proxies and non-proxies if that would be a better option for the future (putting two parameters in the function that default to false then set one or both the true for the desired operation). How does that sound?